### ISSUE TITLE
User or org collective type setting Menus can be accessed for noramal collective two 

### ISSUE THREAD

**Issue by jeffin143** [opencollective/opencollective#3025]
**Describe the bug**
There is a check for the following  menu to be only active when it is ORG type :
https://github.com/opencollective/opencollective-frontend/blob/5755ebac671bab4174ebfc2b8cf3565dd8ab1db6/components/edit-collective/MenuEditCollective.js#L141 
But if you add gift-card in front of a collective url/edit , it will give render the component, I am not sure whether we could add the card, because I am sure internally we would be check again even though the menu is rendered, And it will throw an error 

**To Reproduce**
1. Go to collective
2. Click on edit 
3. append /gift-card to the link
4. See the option of adding a gift card for collective

**Expected behavior**
IG it should not happend, Not sure :) , even though if it happens the functionality won't work I think so 

**Screenshots**
<img width="1680" alt="Screenshot 2020-03-30 at 8 02 13 PM" src="https://user-images.githubusercontent.com/21286134/77925384-6c096b80-72c2-11ea-9285-2b3557a9c738.png">

**Desktop (please complete the following information):**
 - OS: Mac 
 - Browser: Chrome 
 - Version : irrevelant 

**Smartphone (please complete the following information):**

**Additional context**
Not sure if this is a bug, please close it if it isn't, This was while testing for the comment :- https://github.com/opencollective/opencollective-frontend/pull/3794#issuecomment-606025281

**Comment by Betree** [comment]
This is true for all the settings pages. By using the helpers from `components/edit-collective/MenuEditCollective.js` we could show a message instead of the forms in `components/edit-collective/EditCollectiveForm.js` if the collective doesn't have access to this settings page:

> This feature is not available for this collective.
